spec/slavery_spec.rb

```diff
@@ -82,13 +85,30 @@ class ActiveRecord::Relation
     end
   end
 
+  it 'works with nils and blanks' do
+    replica_value = User.on_slave.count
+    expect(User.on_slave("").count).to be replica_value
```

Comment:
Would you explain why we want to support `""`? I think it's a sign of bad caller and should raise IMO. Also it makes our code base fairly complicated.

Reply:
Just handling it like we would handle nil, ie revert to using `:slave` target. Seems safe and since this is parameter is likely to be an expression in calling code, there is at least some possibility that this might evaluate to `""`.

Reply:
Not convinced. Converting strings to symbols should be caller's responsibility.

Reply:
Again, it's about expanding our surface area. You know what I mean if you've tried `"".to_sym` or `nil.to_s` and all that. All those edge cases add up.

Reply:
@kenn let's define clearly what we want the spec for `Slavery.on_slave` to be.

`Slavery.on_slave` => use `slave` target
`Slavery.on_slave(:two)` => use `slave_two` target
`Slavery.on_slave(nil)` => use `slave` target?
`Slavery.on_slave("two")` => raise?
`Slavery.on_slave("")` => raise?

Reply:
`Slavery.on_slave(:slave)` => use `slave` target
`Slavery.on_slave("slave")` => raise?
`Slavery.spec_key = hash; Slavery.on_slave` => use `hash` connection hash
`Slavery.spec_key = hash; Slavery.on_slave(:two)` => use `slave_two` target

Reply:
Ah, good point.

```ruby
Slavery.on_slave => use slave target
Slavery.on_slave(:two) => use slave_two target
Slavery.on_slave(nil) => raise if possible, or use slave target - don't care much, implementation simplicity matters more
Slavery.on_slave("two") => raise
Slavery.on_slave("") => raise
Slavery.on_slave(:slave) => use slave_slave target
Slavery.on_slave("slave") => raise
Slavery.spec_key = 'slave'; Slavery.on_slave => use 'slave' spec
Slavery.spec_key = 'slave'; Slavery.on_slave(:two) => use slave_two target
```

I think deprecating `Slavery.spec_key` at the same time is an option with a major update, as it is inherently incompatible with multiple slave support. It should be replaced something like `spec_key_base`, but I'd leave it out for future discussion.

Reply:
See https://github.com/kenn/slavery/pull/3 for original discussion for `spec_key`.